modules/post/windows/gather/enum_browsers.rb

```diff
@@ -0,0 +1,896 @@
+require 'sqlite3'
+
+IV_SIZE = 12
+TAG_SIZE = 16
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Advanced Browser Data Extraction for Chromium and Gecko Browsers',
+        'Description' => %q{
+          This post-exploitation module extracts sensitive browser data from both Chromium-based and Gecko-based browsers
+          on the target system. It supports the decryption of passwords and cookies using Windows Data Protection API (DPAPI)
+          and can extract additional data such as browsing history, keyword search history, download history, autofill data,
+          credit card information, browser cache and installed extensions.
+        },
+        'License' => MSF_LICENSE,
+        'Platform' => ['win'],
+        'Arch' => [ ARCH_X64, ARCH_X86 ],
+        'Targets' => [['Windows', {}]],
+        'SessionTypes' => ['meterpreter'],
+        'Author' => ['Alexander "xaitax" Hagenah'],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptBool.new('KILL_BROWSER', [false, 'Kill browser processes before extracting data.', false]),
+      OptBool.new('USER_MIGRATION', [false, 'Migrate to explorer.exe running under user context before extraction.', false]),
+      OptEnum.new('BROWSER_TYPE', [true, 'Specify which browser type to extract data from.', 'all', ['all', 'chromium', 'gecko']]),
```

Comment:
Would it be possible to support running this module with a specific browser type? i.e. opera etc

Reply:
Postponed it as long as I possibly could. No excuses anymore, thanks. :)